iio-widgets/src/datastrategy/deviceattrdatastrategy.cpp

```diff
@@ -0,0 +1,57 @@
+#include "datastrategy/deviceattrdatastrategy.h"
+#define BUFFER_SIZE 256
+
+Q_LOGGING_CATEGORY(CAT_DEVICE_DATA_STRATEGY, "DeviceDataStrategy")
+
+using namespace scopy::attr;
+
+DeviceAttrDataStrategy::DeviceAttrDataStrategy(IIOWidgetFactoryRecipe recipe, QObject *parent)
+{
+	m_recipe = recipe;
+	setParent(parent);
```

Comment:
A better approach here would be to create the underlying QObject before the constructor like this:

DeviceAttrDataStrategy::DeviceAttrDataStrategy(IIOWidgetFactoryRecipe recipe, QObject *parent) : QObject(parent)

Please apply this to all QObjects